docs/source/developer/extension_tutorial.rst

```diff
@@ -871,84 +873,64 @@ after the refresh.
    The completed extension, showing the `Astronomy Picture of the Day for 24 Jul 2015 <https://apod.nasa.gov/apod/ap150724.html>`__.
 
 Refer to the `05-restore-panel-state
-tag <https://github.com/jupyterlab/jupyterlab_apod/tree/2.0-05-restore-panel-state>`__
+tag <https://github.com/jupyterlab/jupyterlab_apod/tree/3.0-05-restore-panel-state>`__
 if your extension is not working correctly. Make a commit when the state of your
 extension persists properly.
 
 .. code:: bash
 
-    git add .
+    git add src/index.ts
     git commit -m 'Restore panel state'
 
 Congratulations! You've implemented all of the behaviors laid out at the start
-of this tutorial. Now how about sharing it with the world?
-
-.. _publish-your-extension-to-npmjsorg:
-
-Publish your extension to npmjs.org
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-npm is both a JavaScript package manager and the de facto registry for
-JavaScript software. You can `sign up for an account on the npmjs.com
-site <https://www.npmjs.com/signup>`__ or create an account from the
-command line by running ``npm adduser`` and entering values when
-prompted. Create an account now if you do not already have one. If you
-already have an account, login by running ``npm login`` and answering
-the prompts.
-
-Next, open the project ``package.json`` file in your text editor. Prefix
-the ``name`` field value with ``@your-npm-username>/`` so that the
-entire field reads ``"name": "@your-npm-username/jupyterlab_apod"`` where
-you've replaced the string ``your-npm-username`` with your real
-username. Review the homepage, repository, license, and `other supported
-package.json <https://docs.npmjs.com/files/package.json>`__ fields while
-you have the file open. Then open the ``README.md`` file and adjust the
-command in the *Installation* section so that it includes the full,
-username-prefixed package name you just included in the ``package.json``
-file. For example:
+of this tutorial.
+
+.. _packaging your extension:
+
+Packaging your extension
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+JupyterLab extensions for JupyterLab 3.0 can be distributed as Python
+packages. The cookiecutter template we used contains all of the Python
+packaging instructions in the ``setup.py`` file. Before generating a package, we first need to install ``jupyter_packaging``.
```

Comment:
Should we encourage publishing both so it can be used as a `bundled` extension?

Reply:
I think not, at least in this very first tutorial. The npm publishing instructions were quite intimidating, and it was great to just cut them all out. This tutorial should focus on the simplest recommended case, I think.

I think another tutorial where you have an extension that creates a new token used by other extensions (thereby necessitating an npm publish step), or perhaps just an appendix to this tutorial talking about why you would want to publish to npm, would be good.

Reply:
Yeah, I like the idea of an Appendix, and listing it as an optional step

Reply:
Can you take a look at the text I included now?